scalableminds/webknossos
# Upgrade to s3 client v2

The new s3 java client lib offers an async api :tada: 

### URL of deployed dev instance (used for testing):
- https://newsclient.webknossos.xyz

### Steps to test:
- Open an s3 remote dataset, should load as expected (hopefully now with fewer blocked threads)
- I also added a couple new unit tests
- I also repeated the steps-to-test from https://github.com/scalableminds/webknossos/pull/7453

### TODOs:
- [x] test range request
- [x] test directory listing
- [x] test error handling for directory listing (should be Box.Empty)
- [x] Region
- [x] test with custom endpoint
- [x] Encoding

### Issues:
- fixes  #7913

------
